pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -50,6 +55,13 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 	log.Debugf("[lifecycle] Invocation invokeEvent payload is: %s", startDetails.InvokeEventRawPayload)
 	log.Debug("[lifecycle] ---------------------------------------")
 
+	// Create a new appsec execution context for this invocation
```

Comment:
Here we introduce a tight coupling between lifecycle and the appsec package.
I think it would be nice if we could remove this tight coupling.

Maybe with a `process` method. When lifeCycle is created, let's attach this `process` method on it, and all the logic will live in the appsec package. (feature flag, header manipulation etc.)
This way in `lifecycle.go` we would have something like : 
`lp.process()` which will be totally agnostic from the appSec package.

What do you think?